wazimap_ng/cache.py

```diff
@@ -65,55 +71,86 @@ def last_modified_point_updated(request, profile_id, profile_category_id=None, t
 
     return last_modified(request, profile_id, key)
 
+
 ########### Signals #################
 def update_profile_cache(profile):
     logger.info(f"Updating profile cache: {profile}")
     key = profile_key % profile.id
     cache.set(key, datetime.now())
 
+
 def update_point_cache(category):
     profile = category.profile
     key1 = location_key % (profile.id, category.id)
 
     logger.debug(f"Set cache key (category): {key1}")
     cache.set(key1, datetime.now())
 
-@receiver(post_save, sender=ProfileIndicator)
+
+@receiver(post_save, sender=profile_models.ProfileIndicator)
 def profile_indicator_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=ProfileHighlight)
+
+@receiver(post_save, sender=profile_models.ProfileHighlight)
 def profile_highlight_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=IndicatorCategory)
+
+@receiver(post_save, sender=profile_models.IndicatorCategory)
 def profile_category_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=IndicatorSubcategory)
+
+@receiver(post_save, sender=profile_models.IndicatorSubcategory)
 def profile_subcategory_updated(sender, instance, **kwargs):
     update_profile_cache(instance.category.profile)
 
-@receiver(post_save, sender=ProfileKeyMetrics)
+
+@receiver(post_save, sender=profile_models.ProfileKeyMetrics)
 def profile_keymetrics_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=Profile)
+
+@receiver(post_save, sender=profile_models.Profile)
 def profile_indicator_updated(sender, instance, **kwargs):
     update_profile_cache(instance)
 
-@receiver(post_save, sender=Location)
+
+@receiver(post_save, sender=points_models.Location)
 def point_updated_location(sender, instance, **kwargs):
     update_point_cache(instance.category)
 
-@receiver(post_save, sender=Category)
+
+@receiver(post_save, sender=points_models.Category)
 def point_updated_category(sender, instance, **kwargs):
     update_point_cache(instance)
 
+
+@receiver(post_save, sender=profile_models.Indicator)
+def indicator_updated(sender, instance, **kwargs):
+    indicator_id = instance.id
+    all_profile_indicators = profile_models.ProfileIndicator.objects.filter(
+        indicator_id=indicator_id
+    ).values_list('profile_id', flat=True)
+    all_profile_key_metrics = profile_models.ProfileKeyMetrics.objects.filter(
+        variable_id=indicator_id
+    ).values_list('profile_id', flat=True)
+    all_profile_highlights = profile_models.ProfileHighlight.objects.filter(
+        indicator_id=indicator_id
+    ).values_list('profile_id', flat=True)
+    # find the unique profiles whose cache needs to be invalidated
+    list_of_profile_cache_to_be_update = set(all_profile_indicators + all_profile_key_metrics + all_profile_highlights)
```

Comment:
what about renaming the function name to a more precise name and rename this to `unique_profile_ids_to_invalidate`